src/nupic/algorithms/Connections.cpp

```diff
@@ -362,16 +407,24 @@ void Connections::computeActivity(
     Permanence connectedPermanence) const {
   NTA_ASSERT(numActiveConnectedSynapsesForSegment.size() == segments_.size());
   NTA_ASSERT(numActivePotentialSynapsesForSegment.size() == segments_.size());
+  NTA_CHECK( abs(connectedPermanence - EPSILON - connectedThreshold_) <= EPSILON );
```

Comment:
where do you get this EPSILON? Use nupic::Epsilon from math/Math.hpp

Reply:
EPSILON comes from Connections.hpp line 53.  Its used by the connections class and TM class.
`static const Permanence EPSILON = 0.00001f;`

Compared to Math.hpp:
```
/**
 *  Epsilon is defined for the whole math and algorithms of the Numenta
 * Platform, independently of the concrete type chosen to handle floating point
 * numbers. numeric_limits<float>::epsilon() == 1.19209e-7
 *   numeric_limits<double>::epsilon() == 2.22045e-16
 */
static const nupic::Real Epsilon = nupic::Real(1e-6);
```
I'd rather use  the epsilon which has the correct data type (Permanence is Real32).

Reply:
..we must double check the PR #169 , I think I was fixing (aiming to) this after merge where EPSILON would not be known, and I replaced that with nupic::Epsilon.